.azure_pipelines/ci-pipeline-makefile.yml

```diff
@@ -81,6 +81,15 @@ jobs:
           MYST_SKIP_PR_TEST: 1
           MYST_DO_CLEANUP: 1
 
+      # run PyTorch tests
+      - script: |
```

Comment:
In fact, I think they would be run by default with the other tests as long as you add the line in the solutions Makefile for these

Reply:
@jxyang @salsal97 For now, PyTorch tests run for ~36 minutes. Currently, only 5 core test suites are enabled. We might enable further tests (there are 100+ more).

The main consideration to run it as a separate stage is to boost parallelism: I tend to break down long running stage to multiple smaller ones, so that (1) they can run in different schedules, and (2) the whole pipeline can finish sooner if running in full parallel.

What do you think?

Reply:
Hmm, I think the pipeline stages are serialized in ADO. Not sure about Jenkins though. I did notice this stage takes huge disk space, thus maybe it's advantageous to run it and clean it up separately to save disk space. For example, we could remove the docker image, appdir, and rootfs completely for PyTorch. For the other solution tests, only appdir is removed on the pipeline.

Reply:
PyTorch is run in a standalone pipeline in ADO. Yes, I considered cleaning up disk space by running a `make clean -C -C $(Build.SourcesDirectory)/solutions/pytorch_tests`  after `make`. That will delete `appdir` and `rootfs`.  I don't think PyTorch needs to be run in PR pipeline. So I will delete it from this file. For docker image, the `mystikos/pytorch-test:v1.10.0` is 2.11GB. That's why I didn't prune it after each run. Caching the docker image will save a couple of minutes of time to pulling image from docker hub.